### PR TITLE
docs: Add JSON schema for IDE validation

### PR DESCRIPTION
**Pull request by brabster**
Checklist:

* [x] My organization is added to [USERS.md](https://github.com/argoproj/argo/blob/master/USERS.md) - I'm a consultant

See #4477

- Adds JSON Schema validation as the first IDE support method, with instructions for IDEA and VSCode
- Deletes existing CRD validation approach

### CODE REVIEW

File: docs/ide-setup.md [hunk 1]
```diff
@@ -47,4 +41,29 @@ Add the Argo schema setting `yaml.schemas`:
 
 That's it. Open an Argo YAML file and you should see smarter behaviour, including type errors and context-sensitive autocomplete.
 
-![VScode Example Functionality](assets/vscode-ide-step-4-example-functionality.png)
\ No newline at end of file
+![VScode Example Functionality](assets/vscode-ide-step-4-example-functionality.png)
+
+## Validating Argo YAML using CRD Validation in IntelliJ Ultimate Edition
```
Comment (alexec) [inline docs/ide-setup.md:46]: would it be possible to replace these with the JSON schema instructions for IntelliJ?

**Review by brabster — COMMENTED** [review]
No review necessary from me

**Comment by alexec** [comment]
Still WIP?

**Review by alexec — APPROVED** [review]
LGTM

**Comment by alexec** [comment]
@brabster can you do the DCO please?

**Comment by brabster** [comment]
Sorted sorry hadn't seen DCO before! Also sorted WIP label etc.

**Comment by alexec** [comment]
LGTM!

**Comment by simster7** [comment]
Back-ported to v2.12.0-RC5

**Pull request merged**